### ISSUE TITLE
"{" Character Removed When Adding Dataset via API

### ISSUE THREAD

**Issue by pickuse2013** [langgenius/dify#11868]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.14.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

When attempting to update documents through the Knowledge API, the "{" character in the input text is incorrectly removed.

This issue occurred after updating from version `v0.13.2` to `v0.14.1`.

**Affected API:**
`/datasets/{dataset_id}/documents/{document_id}/update-by-text`

**Steps to Reproduce:**
1. Use the following input data to call the API:
```json
{
    "name": "test",
    "text": "{aaaa}",
    "indexing_technique": "high_quality",
    "process_rule": {
        "mode": "custom",
        "rules": {
            "pre_processing_rules": [
                {
                    "id": "remove_extra_spaces",
                    "enabled": true
                },
                {
                    "id": "remove_urls_emails",
                    "enabled": false
                }
            ],
            "segmentation": {
                "separator": "\n",
                "max_tokens": 10000,
                "chunk_overlap": 100
            }
        }
    }
}
```
2. Check the stored document in the dataset.

### ✔️ Expected Behavior

The text in the document should remain as `{aaaa}`.

### ❌ Actual Behavior

The stored text in the document becomes `aaaa}`.

**Comment by Augus** [comment]
@takatost  or anyone, could you take a look at this issue? This is a critical problem—updating the document leads to a loss of content integrity. The issue occurs when upgrading from v0.13 to v0.14.1. Based on the latest release notes, it seems that it hasn't been fixed in v0.14.2 either.

**Comment by yihong0618** [comment]
will take a look maybe today

**Comment by zandko** [comment]
> @takatost or anyone, could you take a look at this issue? This is a critical problem—updating the document leads to a loss of content integrity. The issue occurs when upgrading from v0.13 to v0.14.1. Based on the latest release notes, it seems that it hasn't been fixed in v0.14.2 either.
At first, it was to reduce the possibility of such situations occurring in the titles, which would affect the retrieval quality of large models.
